File: src/main/java/com/cfgglobal/generator/task/processor/ITaskProcessor.java
package com.cfgglobal.generator.task.processor;


import com.cfgglobal.generator.entity.CodeProject;
import com.cfgglobal.generator.entity.Task;

import java.util.List;
import java.util.Map;

public interface ITaskProcessor {
    List<String> run(CodeProject codeProject, Task task, Map<String, Object> context);
}

File: src/main/kotlin/com/cfgglobal/generator/task/processor/ITaskProcessor.kt
package com.cfgglobal.generator.task.processor


import com.cfgglobal.generator.entity.CodeProject
import com.cfgglobal.generator.entity.Task

interface ITaskProcessor {
    fun run(codeProject: CodeProject, task: Task, context: MutableMap<String, Any>): List<String>
}
